LibVLCSharp/Shared/Media.cs

```diff
@@ -515,16 +509,53 @@ MediaEventManager EventManager
         /// <returns>duration of media item or -1 on error</returns>
         public long Duration => Native.LibVLCMediaGetDuration(NativeReference);
 
-        /// <summary>Parse a media.
-        /// This fetches (local) art, meta data and tracks information.
-        /// The method is synchronous.
-        /// This function could block indefinitely.
-        /// Use libvlc_media_parse_with_options() instead
-        /// libvlc_media_parse_with_options
-        /// libvlc_media_get_meta
-        /// libvlc_media_get_tracks_info
+        /// <summary>
+        /// Parse the media asynchronously with options.      
+        /// It uses a flag to specify parse options (see <see cref="MediaParseOptions"/>). All these flags can be combined. By default, media is parsed if it's a local file.
+        /// <para/> Note: Parsing can be aborted with ParseStop().
         /// </summary>
-        public void Parse() => Native.LibVLCMediaParse(NativeReference);
+        /// <param name="options">parse options</param>
+        /// <param name="timeout">maximum time allowed to preparse the media. 
+        /// <para/>If -1, the default "preparse-timeout" option will be used as a timeout. 
+        /// <para/>If 0, it will wait indefinitely. If > 0, the timeout will be used (in milliseconds). 
+        /// </param>
+        /// <param name="cancellationToken">token to cancel the operation</param>
+        /// <returns>the parse status of the media</returns>
+        public async Task<MediaParsedStatus> Parse(MediaParseOptions options = MediaParseOptions.ParseLocal, int timeout = -1, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            
+            TaskCompletionSource<MediaParsedStatus> tcs = null;
+            var cancellationTokenRegistration = cancellationToken.Register(() =>
+            {
+                ParsedChanged -= OnParsedChanged;
+                Native.LibVLCMediaParseStop(NativeReference);
+                tcs?.TrySetCanceled();
+            });
+
+            void OnParsedChanged(object sender, MediaParsedChangedEventArgs mediaParsedChangedEventArgs) 
+                => tcs?.TrySetResult(mediaParsedChangedEventArgs.ParsedStatus);
+
+            try
+            {
+                tcs = new TaskCompletionSource<MediaParsedStatus>();
+                        
+                ParsedChanged += OnParsedChanged;
+
+                var result = Native.LibVLCMediaParseWithOptions(NativeReference, options, timeout);
+                if (result == -1)
+                {
+                    tcs.TrySetResult(ParsedStatus);
```

Comment:
ParsedStatus will not be initialized if IsParsed is `false` (i.e., it will have its 0 default value, which is not a valid enum value).

`IsParsed` will likely be `false`, because the parse method will not touch it, and it will not be parsed before parsing...

Instead, I'd say that you should return `MediaParsedStatus.Failed` instead, because well, it's a failure anyway...